tickit/adapters/interpreters/endpoints/http_endpoint.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import AnyStr, Callable, Generic
+
+from aiohttp.web_routedef import RouteDef
+
+
+@dataclass(frozen=True)
+class HTTPEndpoint(Generic[AnyStr]):
```

Comment:
@garryod and @thomascobb, naming debate time, what do we prefer?

- `HTTPEndpoint`
- `RestEndpoint`
- `Endpoint`
- `Put`/`Get`/etc.

Something else?

Reply:
I like `HTTPEndpoint`(assuming our implementation complies with the HTTP spec) as the others are all more generic (despite REST often being used interchangeably).

Reply:
Something like that would be nicer, yes, although I'm hesitant about passing `kwargs` as it's nice to have completion in IDE when filling in these decorators. We could subclass it:

```python
class GetEndpint(HTTPEndpoint):
    method: str = "GET"

### LATER ON

@GetEndpoint("/foo")
def foo(self, request: Request) -> Response:
    ...
```

What do you think?

Reply:
Sounds fine to me

Reply:
We decided on your original idea, @thomascobb because with mine you could do:
```python
@GetEndpoint(method="PUT")
def foo(...)
```

Reply:
ok, `**kwargs` was just a shortcut for "the rest of the args with their types", it's quite a bit of duplication, but at least it's in library code rather than device code